node/as/thrift.js

```diff
@@ -53,8 +55,29 @@ function TChannelAsThrift(opts) {
     self.logParseFailures = typeof logParseFailures === 'boolean' ?
         logParseFailures : true;
 
-    // only used in request()
     self.channel = opts.channel;
+
+    self.healthCheckCallback = opts.healthCheckCallback;
+    assert(!self.healthCheckCallback || typeof self.healthCheckCallback === 'function',
+        'healthCheckCallback must be a function');
+    assert(!self.healthCheckCallback || self.channel,
+        'channel must be provided with healthCheckCallback');
```

Comment:
Let’s be sure to note that this is a backward incompatible change in the migration docs, and make a major version bump, or not make a breaking change.